contracts/factory/src/contract.rs

```diff
@@ -174,6 +187,10 @@ impl FactoryTrait for Factory {
         )
             .into_val(&env);
 
+        if let Some(amp) = amp {
+            init_fn_args.push_back(amp.into_val(&env));
+        }
```

Comment:
Bug spotted. We are allowing AMP parameter, in case it's an XYK pool we just ignored it.
In that case, this  would crash the call.

```suggestion
        if let PoolType::Stable = pool_type {
            init_fn_args.push_back(amp.into_val(&env));
        }
```

Reply:
Plus it shows you didn't test that path!

Reply:
added the missing test cases